src/states/infiniteweightpeps.jl

```diff
@@ -0,0 +1,242 @@
+
+"""
+    const PEPSWeight{S}
+
+Default type for PEPS bond weights with 2 virtual indices,
+conventionally ordered as: ``wt : ES ← WN``. 
+Here, `ES`, `WN` denote the east/south, west/north spaces, respectively.
+"""
+const PEPSWeight{S} = AbstractTensorMap{S,1,1} where {S<:ElementarySpace}
+
+"""
+Schmidt bond weight used in simple/cluster update
+"""
+struct SUWeight{E<:PEPSWeight}
+    x::Matrix{E}
+    y::Matrix{E}
+
+    function SUWeight(x::Matrix{E}, y::Matrix{E}) where {E<:PEPSWeight}
+        if size(x) != size(y)
+            throw(
+                ArgumentError(
+                    "Matrices for x-weights and y-weights must have the same size, but got size(x) = $(size(x)) and size(y) = $(size(y)).",
+                ),
+            )
+        end
+        return new{E}(x, y)
+    end
+end
+
+function Base.size(wts::SUWeight)
+    return size(wts.x)
+end
+
+function Base.eltype(wts::SUWeight)
+    return eltype(wts.x)
+end
+
+function Base.:(==)(wts1::SUWeight, wts2::SUWeight)
+    return wts1.x == wts2.x && wts1.y == wts2.y
+end
+
+function Base.:(+)(wts1::SUWeight, wts2::SUWeight)
+    return SUWeight(wts1.x + wts2.x, wts1.y + wts2.y)
+end
+
+function Base.:(-)(wts1::SUWeight, wts2::SUWeight)
+    return SUWeight(wts1.x - wts2.x, wts1.y - wts2.y)
+end
+
+function Base.show(io::IO, wts::SUWeight)
+    N1, N2 = size(wts)
+    for (direction, r, c) in Iterators.product("xy", 1:N1, 1:N2)
+        println(io, "$direction[$r,$c]: ")
+        wt = (direction == 'x' ? wts.x[r, c] : wts.y[r, c])
+        for (k, b) in blocks(wt)
+            println(io, k, " = ", diag(b))
+        end
+    end
+end
+
+function Base.iterate(wts::SUWeight, state...)
+    return iterate(Iterators.flatten((wts.x, wts.y)), state...)
+end
+
+function Base.length(wts::SUWeight)
+    return 2 * prod(size(wts.x))
+end
+
+function Base.isapprox(wts1::SUWeight, wts2::SUWeight; atol=0.0, rtol=1e-5)
+    return (
+        isapprox(wts1.x, wts2.x; atol=atol, rtol=rtol) &&
+        isapprox(wts1.y, wts2.y; atol=atol, rtol=rtol)
+    )
+end
+
+function compare_weights(wts1::SUWeight, wts2::SUWeight)
+    wtdiff = sum(_singular_value_distance((wt1, wt2)) for (wt1, wt2) in zip(wts1, wts2))
+    return wtdiff / (2 * prod(size(wts1)))
```

Comment:
Here you could rewrite this to not use iteration instead, which gives the same result. I'm not entirely convinced that iterating over weights is an intuitive operation, so I would prefer to have `length` return the indexing length, ie the sites in the unitcell
```suggestion
    (L = length(wts1)) == length(wts2) || throw(ArgumentError("size mismatch"))
    return (sum(_singular_value_distance, zip(wt1.x, wt2.x)) + sum(_singular_value_distance, zip(wt1.y, wt2.y))) / (2L)
```

Reply:
The main problem is that `size` and `length` are incompatible, which is a bit surprising to a user (and might lead to strange bugs when combining broadcasting/iterating/map/.... 

One alternative solution could be that `size = (2, rows, cols)`, to indicate that you have `dir, row, col` for the weights. (you could even define `getindex` and `setindex!` to reflect that behavior). That way, this ambiguity is resolved, and you still have iteration, which I would probably define as:
```julia
Base.iterate(wt, state=0) = iseven(state) ? wt.x[div(state, 2)] : wt.y[div(state, 2)]
```
Which would then return the x and y weights interleaved, but that is more consistent with the `dir, row, col` convention we have throughout PEPSKit.